### PR TITLE
converter: implement SameObject

### PR DESCRIPTION
**Pull request by gabrielschulhof**
Generate idempotent versions of property getter results for attributes
marked as `SameObject`.

### CODE REVIEW

File: webidl-napi-inl.h [hunk 1]
```diff
@@ -435,5 +428,74 @@ inline napi_ref InstanceData::GetConstructor(const char* name) {
...
+    wrapping->refs.resize(same_obj_count, nullptr);
+  return napi_wrap(env, js_rcv, wrapping, Destroy, nullptr, nullptr);
+}
+
+// static
+template <typename T>
+napi_status Wrapping<T>::Retrieve(napi_env env,
+                                  napi_value js_rcv,
+                                  T** cc_rcv,
+                                  int ref_idx,
```
Comment (kainino0x) [inline webidl-napi-inl.h:449]: optional suggestion: I think I eventually figured out how it worked, but this system is pretty hard to follow. Seems like you assign a `ref_idx` to each member that has `[SameObject]`, and then store the ref for that value in `refs[ref_idx]` for that member. Could it be possible to have the `refs` members instead be actual named `napi_ref` members in a generated struct?

Current code looks reasonably correct though.

**Review by kainino0x — APPROVED** [review]

**Comment by gabrielschulhof** [comment]
Landed in 88a9f9aa6975351b52b4a787468185c8a5024685.

File: webidl-napi-inl.h (hunk 1 above)
Comment (gabrielschulhof) [reply webidl-napi-inl.h:449]: I could go with a std::map instead.

Comment (kainino0x) [reply webidl-napi-inl.h:449]: If the map would be keyed on a string, I think the current code is better.
